src/routes/api/db-migrations/db-migration-utils.ts

```diff
@@ -0,0 +1,91 @@
+import { Request } from "express";
+import fs from "fs";
+import path from "path";
+import { isNodeDev, isNodeTest, isNodeProd, getNodeEnv } from "utils/is-node-env";
+import { getLogger } from "config/logger";
+import { exec as execOrigin } from "child_process";
+import { promisify } from "util";
+
+const exec = promisify(execOrigin);
+const logger = getLogger("DBMigration");
+
+/**
+ *
+ * Functiont to get the parem for db migration.
+ *
+ * Intentially to NOT make this param case insensitive to be more safe
+ *
+ * @returns The common param in req.body `targetScript` to migrate up or down
+ */
+export const getTargetScript = (req: Request) => {
+	const targetScript = (req.body || {}).targetScript;
+	if (!targetScript) {
+		throw {
+			statusCode: 400,
+			message: `"targetScript" is mandatory in the request body, but found none.`
+		};
+	}
+	return targetScript;
+};
+
+/* Make sure the script to migrate up or down is the latest script in this build.
+ * So the following scenarios will fail.
+ * Person A merge script 1, person B merge scripts 2. Now we CAN NOT migrate anymore. Need to revert one PR first to migrate.
+ *
+ * In short, encourage ppl to over-communicate when it comes to DB migrations.
+ */
+export const validateScriptLocally = async (targetScript: string) => {
+
+	const scripts = await fs.promises.readdir(path.resolve(process.cwd(), "db/migrations"));
+	//Sort by name, asc, so filename order has to be in order now.
+	//So this now will be mandatory that, db migrtions scripts has to be alphabetically ordered.
+	scripts.sort();
+	const latestScriptsInRepo = scripts[scripts.length-1];
+
+	if (targetScript !== latestScriptsInRepo) {
+		throw {
+			statusCode: 400,
+			message: `"targetScript: ${targetScript}" doesn't match latest scripts in db/migrations ${latestScriptsInRepo}`
+		};
+	}
+
+	logger.info(`Target script match latest scripts in repo ${targetScript}, validation passed`);
+
+};
+
+export enum DBMigrationType {
+	UP = "UP",
+	DOWN = "DOWN"
+}
+
+export const startDBMigration = async (targetScript: string, ops: DBMigrationType) => {
+	const env = getDBMigrateEnv();
+	if (ops !== DBMigrationType.UP && ops !== DBMigrationType.DOWN) {
```

Comment:
This validation needs to be done somewhere else, where the string is converted to enum. Once it is converted, type-safety should guarantee that this never happens.

Reply:
`string is converted to enum` <-- there's no string converted to enum in the code. The code is using it as enum directly.

But but, I just make sure it is safe, as at the end of day, js doesn't have enum, just string.

Reply:
Ah right, I didn't paid attention to it. Anyway, I find it awkward that we are doing validation of something that is always true. 

If we worry that someone might add a new DBMigrationType that won't be supported by this function (which means the ternary operator will fallback to "down"), I would've changed that to 
```
switch (ops) {
   case UP: blah
   case DOWN: yada
   default: throw "Should never happen"
} 
```